packages/nimble-components/src/table/specs/table-column-sort-hld.md

```diff
@@ -0,0 +1,75 @@
+
+A column within the table may not have a one-to-one mapping to a field within the table's data. For example, a hyperlink column might use two fields from the data: a string to use as the hyperlink's display and a string to use as the hyperlink's href. Because of this, a column must define which data field is used when sorting that column. That field is specified by a column definition by implementing an `operationalDataFieldName` getter that returns the name of the field to use for sorting. For example, the text column would return the value of `fieldName` and the hyperlink column would return the field name associated with the display string.
+
+For performance reasons, the table will not support custom sort functions on a column. A column can only be sorted based on a single field within the table's data, which guards against performance degradations from inefficient custom sort functions.
+
+#### Summary of new attributes on the base table column
+
+| attribute name | type                                                                                                         | default value                   | description                                                     |
+| -------------- | ------------------------------------------------------------------------------------------------------------ | ------------------------------- | --------------------------------------------------------------- |
+| sort-direction | `TableColumnSortDirection`, defined as `{none: undefined, ascending: 'ascending', descending: 'descending'}` | `TableColumnSortDirection.none` | The direction the column is sorted                              |
+| sort-index     | `number` or `null` \*                                                                                        | `null`                          | The direction the column is sorted                              |
+| hidden         | `boolean`                                                                                                    | `false`                         | When set to true, do not render the column as part of the table |
+
+\* Note: The `sort-index` attribute is `number | null` because of the plan to use the `nullableNumberConverter` provided by FAST. That converter uses the value of `null` to represent non-number types rather than `undefined`, which is common within the nimble repo.
+
+### Validation
+
+The table will be invalid if it contains two columns with the same `sort-index`. If a duplicate `sort-index` is detected, the rows in the table will not render, no sort indicators will render, and the table's validity object will have `duplicateColumnSortIndex` set to `true`.
```

Comment:
Is this level of rigidity necessary? I would think we treat identical sort indexes the same way any sort routine would treat identical values.

Reply:
I could be convinced either way, but I tend to think that if we have a state that would lead to non-deterministic behavior, it should be invalid. Since having identical `sort-index` values would lead to non-deterministic behavior of what order we pass those columns to TanStack, I think we should consider it invalid. Though, the part of me that could be convinced otherwise, says that nothing will technically go wrong if the columns are sorted in a non-deterministic order.

I'd be curious to get the input of others -- @rajsite @jattasNI